packages/common/src/services.ts

```diff
@@ -721,6 +721,34 @@ export class CondaPackage implements Conda.IPackageManager {
     }
   }
 
+  async getDependencies(
+    pkg: string,
+    cancellable = true
+  ): Promise<Conda.IPackageDeps> {
+    this._cancelTasks();
```

Comment:
You are introducing a second cancellable action. So it would be good to change a bit the API to store in the cancellable stack objects `{type, cancel}`. So `_cancelTasks` can be called on a particular `type` of tasks.

Meaning here:
```suggestion
    this._cancelTasks('getDependencies');
```